### PR TITLE
Pull Request 5/3/2020

### PR DESCRIPTION
**Pull request by nateelee**
fixed firebase call issue where username is undefined

reorganized code in app.js so that firebase calls will be used under certain conditions

modified mutator function for username data purposes

implement an editor with a movie search and upload (beta)

### CODE REVIEW

File: src/components/profile.jsx [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-        profilePicUrl: '',
-        bio: ''
+    constructor(props) {
+    super(props)
+       this.state = { 
+            id: '',
+            username: '',
+            profilePicUrl: '',
+            bio: ''
+        }
```
Comment (pkuang5) [inline src/components/profile.jsx]: i dont think you changed anything in this file. can you revert back to original state

File: src/components/googleLogInBtn.jsx [hunk 2]
```diff
@@ -29,8 +29,13 @@ class GoogleLogInBtn extends Component {
 
         });
       }
+      else {
+        username = snapshot.val().userName
+        console.log(username)
+      }
+      this.props.signInState(true, googleId, username);
```
Comment (pkuang5) [inline src/components/googleLogInBtn.jsx]: can u explain this for me

File: src/components/feed.jsx [hunk 3]
```diff
@@ -13,6 +11,7 @@ class Feed extends Component {
         var userInfo = firebase.database().ref('users/' + this.props.googleId)
         userInfo.on('value', (snapshot) => {
             if (snapshot.val()) this.setState({firstName: snapshot.val().firstName})
+            console.log(snapshot.val())
```
Comment (pkuang5) [inline src/components/feed.jsx]: delete

File: src/App.js [hunk 4]
```diff
@@ -60,10 +55,11 @@ class App extends Component {
           <Navbar signInState={this.signInState} username={this.state.username} />
           <Switch>
             <Route path="/" exact strict component={() => <Feed signInState={this.signInState} googleId={this.state.googleId} />}></Route>
-            <Route path="/editor" exact strict component={Editor}></Route>
+            <Route path="/editor" exact strict component={() => <Editor signInState={this.signInState} googleId={this.state.googleId} />}></Route>
```
Comment (pkuang5) [inline src/App.js]: how come u pass sign in state to editor

Comment (nateelee) [reply src/App.js]: my mistake.

File: src/components/googleLogInBtn.jsx (hunk 2 above)
Comment (nateelee) [reply src/components/googleLogInBtn.jsx]: so basically, after debugging last night, i found that the last issue was that upon the first render username is unidentified. and the same goes for googleId, and the way we passed !null values was through signInState at the very beginning upon successful login.

File: src/components/profile.jsx (hunk 1 above)
Comment (nateelee) [reply src/components/profile.jsx]: oh it doesn't. can you explain what the purpose of constructor(props) and super(props) is then? I thought those were needed to pass props.

Comment (pkuang5) [reply src/components/profile.jsx]: honestly i have no idea what thats for. u can still access props by doing this.props. i think u use that when you want to change the value of the props in the child class but i could be wrong

**Comment by pkuang5** [comment]
LGTM

**Pull request merged**